public/resources/core/js/controllers/product-offering.controller.js

```diff
@@ -113,19 +117,51 @@
 
         vm.pricePlan = new Offering.PricePlan();
         vm.pricePlanEnabled = false;
+        vm.priceAlteration = 0;
 
         vm.createPricePlan = createPricePlan;
         vm.updatePricePlan = updatePricePlan;
         vm.removePricePlan = removePricePlan;
 
+        vm.togglePriceAlteration = togglePriceAlteration;
+
         vm.place = "";
         vm.places = [];
 
         vm.createPlace = createPlace;
         vm.removePlace = removePlace;
 
         $scope.$on(Offering.EVENTS.PRICEPLAN_UPDATED, function (event, index, pricePlan) {
-            angular.merge(vm.data.productOfferingPrice[index], pricePlan);
+            angular.extend(vm.data.productOfferingPrice[index], pricePlan);
+        });
+
+        Catalogue.search({
+            owner: true,
+            status: LIFECYCLE_STATUS.LAUNCHED
+        }).then(function (collection) {
+            if (collection.length) {
+                if (collection.length === 1) {
+                    vm.catalogue = collection[0];
+                }
+                return ProductSpec.search({
+                    owner: true,
+                    status: LIFECYCLE_STATUS.LAUNCHED
+                });
+            } else {
+                return $q.reject('Sorry! In order to create a product offering, you must first launch at least one product catalogue.');
+            }
+        }).then(function (collection) {
+            if (collection.length) {
+                if (collection.length === 1) {
+                    vm.product = collection[0];
+                }
+                vm.status = DATA_STATUS.LOADED;
+            } else {
+                return $q.reject('Sorry! In order to create a product offering, you must first create at least one product specification.');
+            }
+        }).catch(function (response) {
```

Comment:
Maybe both then handlers can be replaced by an auxiliary function since they are almost the same